onionperf/measurement.py

```diff
@@ -325,16 +326,29 @@ def __start_tor(self, name, control_port, socks_port, hs_port_mapping=None):
         logging.info("Starting Tor {0} process with ControlPort={1}, SocksPort={2}...".format(name, control_port, socks_port))
 
         tor_datadir = "{0}/tor-{1}".format(self.datadir_path, name)
+        bridge_lines=self.bridge_lines
+
         if not os.path.exists(tor_datadir): os.makedirs(tor_datadir)
 
-        tor_config_template = "ORPort 0\nDirPort 0\nControlPort {0}\nSocksPort {1}\nSocksListenAddress 127.0.0.1\nClientOnly 1\n\
-WarnUnsafeSocks 0\nSafeLogging 0\nMaxCircuitDirtiness 60 seconds\nUseEntryGuards 0\nDataDirectory {2}\nLog INFO stdout\n"
-        tor_config = tor_config_template.format(control_port, socks_port, tor_datadir)
+        base_config_template = "ORPort 0\nDirPort 0\nControlPort {0}\nSocksPort {1}\nClientOnly 1\nSafeLogging 0\nMaxCircuitDirtiness 60 seconds\nDataDirectory {2}\nLog INFO stdout\n".format(control_port, socks_port, tor_datadir)
+
+        if not bridge_lines:
+            tor_config_client = base_config_template + "UseEntryGuards 0"
```

Comment:
Move `tor_config_server = ` above this and set `tor_config_client = tor_config_server` if bridge lines are not present?

Reply:
see commit 7b5e173